lib/App/Pinto/Command/verify.pm

```diff
@@ -15,32 +15,194 @@ use base 'App::Pinto::Command';
 
 #-----------------------------------------------------------------------------
 
+sub opt_spec {
+    my ( $self, $app ) = @_;
+
+    return (
+        [ 'all|a'             => 'Verify everything in the repository'],
+        [ 'authors|A=s'       => 'Limit to matching author identities' ],
+        [ 'distributions|D=s' => 'Limit to matching distribution names' ],
+        [ 'packages|P=s'      => 'Limit to matching package names' ],
+        [ 'pinned!'           => 'Limit to pinned packages (negatable)' ],
+        [ 'stack|s=s'         => 'Limit to contents of this stack' ],
+        [ 'strict'            => 'Make verification warnings fatal' ],
+        [ 'files-only'        => 'Skip crytographic checks' ],
+    );
+}
+
+#------------------------------------------------------------------------------
+
+sub validate_args {
+    my ( $self, $opts, $args ) = @_;
+
+    $self->usage_error('Multiple arguments are not allowed')
+        if @{$args} > 1;
+
+    $opts->{format} = interpolate( $opts->{format} )
+        if exists $opts->{format};
+
+    $opts->{stack} = $args->[0]
+        if $args->[0];
+
+    $self->usage_error('Cannot specify a stack when using --all')
+        if $opts->{stack} && $opts->{all};
+
+    return 1;
+}
+
+#------------------------------------------------------------------------------
+
 1;
 
 __END__
 
 =head1 SYNOPSIS
 
-  pinto --root=REPOSITORY_ROOT verify
+  pinto --root=REPOSITORY_ROOT verify [OPTIONS]
```

Comment:
For commands that can take a stack name as an argument like `list` and `log` I think we also show that in the `SYNOPSIS`.